src/net.rs

```diff
@@ -493,7 +527,7 @@ impl<S: NetworkStream> NetworkStream for HttpsStream<S> {
 
 /// A Http Listener over SSL.
 #[derive(Clone)]
-pub struct HttpsListener<S: Ssl> {
+pub struct HttpsListener<S: SslServer> {
```

Comment:
I don't believe so. You don't need to have the trait imported for a type to implement it.